src/huggingface_hub/HfRepostiory.py

```diff
@@ -0,0 +1,138 @@
+import os
+import subprocess
+from typing import Dict, List
+
+
+MODEL_CARD_TEMPLATE = """
+---
+tags:
+- sagemaker
+datasets:
+- {dataset}
+---
+## {model_id} Trained from SageMaker HuggingFace extension.
+
+#### Hyperparameters
+```json
+{hyperparameters}
+```
+
+#### Eval
+| key | value |
+| --- | ----- |
+{eval_table}
+"""
+
+
+class HfRepository:
+    """Git-based system for HuggingFace Hub repositories"""
+
+    def __init__(
+        self,
+        repo_url: str,
+        huggingface_token: str,
+        model_dir: str,
+    ):
+        self.repo_url = self.add_token_to_repository_url(repo_url, huggingface_token)
+
+        self.model_dir = model_dir
+
+    def init_new_repository(
```

Comment:
`init_new_repository` is not the right name IMO because it doesn't actually `git init` just clones